TestEnrollAndLog/TestEnroll  test is flaky

## Flaky Test

* **Test Name:** TestEnrollAndLog/TestEnroll 
* **Link:** https://github.com/elastic/elastic-agent/blob/123579115ad7a330779c2f9b5b9e719f98282dc4/testing/integration/enroll_test.go#L117
* **Branch:** [Git branch the test was seen in. If a PR, the branch the PR was based off.](https://github.com/elastic/elastic-agent/pull/3071)

### Stack Trace

```
enroll_test.go:189: --- debugging: results from ES --- END ---
    enroll_test.go:106: metricbeat: Got 10 documents
    enroll_test.go:109: Making sure there are no error logs
    enroll_test.go:185: --- debugging: results from ES --- START ---
    enroll_test.go:187: map[string]interface {}{"@timestamp":"2023-07-12T23:58:08.805Z", "address":"127.0.0.1:9200", "agent":map[string]interface {}{"ephemeral_id":"27b2ee5c-a601-412c-8dd9-bf069ecaf1ea", "id":"8129be2d-bd95-412f-a91e-1d94a8c29764", "name":"ogc-cinjkvvqfsgpnvqf9lr0-3011", "type":"filebeat", "version":"8.10.0"}, "cloud":map[string]interface {}{"account":map[string]interface {}{"id":"elastic-platform-ingest"}, "availability_zone":"us-central1-a", "instance":map[string]interface {}{"id":"2851333541445403722", "name":"ogc-cinjkvvqfsgpnvqf9lr0-3011"}, "machine":map[string]interface {}{"type":"t2a-standard-2"}, "project":map[string]interface {}{"id":"elastic-platform-ingest"}, "provider":"gcp", "region":"us-central1", "service":map[string]interface {}{"name":"GCE"}}, "component":map[string]interface {}{"binary":"metricbeat", "dataset":"elastic_agent.metricbeat", "id":"system/metrics-default", "type":"system/metrics"}, "data_stream":map[string]interface {}{"dataset":"elastic_agent.metricbeat", "namespace":"q49ily4xofxhfz0rsujmobuotbwk5yer7hfczxcrs", "type":"logs"}, "ecs":map[string]interface {}{"version":"8.0.0"}, "elastic_agent":map[string]interface {}{"id":"8129be2d-bd95-412f-a91e-1d94a8c29764", "snapshot":true, "version":"8.10.0"}, "event":map[string]interface {}{"agent_id_status":"verified", "dataset":"elastic_agent.metricbeat", "ingested":"2023-07-12T23:58:15Z"}, "host":map[string]interface {}{"architecture":"aarch64", "containerized":false, "hostname":"ogc-cinjkvvqfsgpnvqf9lr0-3011", "id":"bb1f9231036ddf511f603b0dac92e7bc", "ip":[]interface {}{"10.128.0.51", "fe80::4001:aff:fe80:33"}, "mac":[]interface {}{"42-01-0A-80-00-33"}, "name":"ogc-cinjkvvqfsgpnvqf9lr0-3011", "os":map[string]interface {}{"codename":"jammy", "family":"debian", "kernel":"5.19.0-1027-gcp", "name":"Ubuntu", "platform":"ubuntu", "type":"linux", "version":"22.04.2 LTS (Jammy Jellyfish)"}}, "input":map[string]interface {}{"type":"filestream"}, "log":map[string]interface {}{"file":map[string]interface {}{"path":"/opt/Elastic/Agent/data/elastic-agent-b3f64a/logs/elastic-agent-20230712.ndjson"}, "offset":59544, "source":"system/metrics-default"}, "log.level":"error", "log.logger":"esclientleg", "log.origin":map[string]interface {}{"file.line":38, "file.name":"transport/logging.go"}, "message":"Error dialing dial tcp 127.0.0.1:9200: connect: connection refused", "network":"tcp", "service.name":"metricbeat"}
    enroll_test.go:189: --- debugging: results from ES --- END ---
    enroll_test.go:113: errors: Got 1 documents
    enroll_test.go:115: map[string]interface {}{"@timestamp":"2023-07-12T23:58:08.805Z", "address":"127.0.0.1:9200", "agent":map[string]interface {}{"ephemeral_id":"27b2ee5c-a601-412c-8dd9-bf069ecaf1ea", "id":"8129be2d-bd95-412f-a91e-1d94a8c29764", "name":"ogc-cinjkvvqfsgpnvqf9lr0-3011", "type":"filebeat", "version":"8.10.0"}, "cloud":map[string]interface {}{"account":map[string]interface {}{"id":"elastic-platform-ingest"}, "availability_zone":"us-central1-a", "instance":map[string]interface {}{"id":"2851333541445403722", "name":"ogc-cinjkvvqfsgpnvqf9lr0-3011"}, "machine":map[string]interface {}{"type":"t2a-standard-2"}, "project":map[string]interface {}{"id":"elastic-platform-ingest"}, "provider":"gcp", "region":"us-central1", "service":map[string]interface {}{"name":"GCE"}}, "component":map[string]interface {}{"binary":"metricbeat", "dataset":"elastic_agent.metricbeat", "id":"system/metrics-default", "type":"system/metrics"}, "data_stream":map[string]interface {}{"dataset":"elastic_agent.metricbeat", "namespace":"q49ily4xofxhfz0rsujmobuotbwk5yer7hfczxcrs", "type":"logs"}, "ecs":map[string]interface {}{"version":"8.0.0"}, "elastic_agent":map[string]interface {}{"id":"8129be2d-bd95-412f-a91e-1d94a8c29764", "snapshot":true, "version":"8.10.0"}, "event":map[string]interface {}{"agent_id_status":"verified", "dataset":"elastic_agent.metricbeat", "ingested":"2023-07-12T23:58:15Z"}, "host":map[string]interface {}{"architecture":"aarch64", "containerized":false, "hostname":"ogc-cinjkvvqfsgpnvqf9lr0-3011", "id":"bb1f9231036ddf511f603b0dac92e7bc", "ip":[]interface {}{"10.128.0.51", "fe80::4001:aff:fe80:33"}, "mac":[]interface {}{"42-01-0A-80-00-33"}, "name":"ogc-cinjkvvqfsgpnvqf9lr0-3011", "os":map[string]interface {}{"codename":"jammy", "family":"debian", "kernel":"5.19.0-1027-gcp", "name":"Ubuntu", "platform":"ubuntu", "type":"linux", "version":"22.04.2 LTS (Jammy Jellyfish)"}}, "input":map[string]interface {}{"type":"filestream"}, "log":map[string]interface {}{"file":map[string]interface {}{"path":"/opt/Elastic/Agent/data/elastic-agent-b3f64a/logs/elastic-agent-20230712.ndjson"}, "offset":59544, "source":"system/metrics-default"}, "log.level":"error", "log.logger":"esclientleg", "log.origin":map[string]interface {}{"file.line":38, "file.name":"transport/logging.go"}, "message":"Error dialing dial tcp 127.0.0.1:9200: connect: connection refused", "network":"tcp", "service.name":"metricbeat"}
    enroll_test.go:117: 
        	Error Trace:	/home/ubuntu/agent/testing/integration/enroll_test.go:117
        	Error:      	Should be empty, but was [{.ds-logs-elastic_agent.metricbeat-q49ily4xofxhfz0rsujmobuotbwk5yer7hfczxcrs-2023.07.12-000001 4.4011974 map[@timestamp:2023-07-12T23:58:08.805Z address:127.0.0.1:9200 agent:map[ephemeral_id:27b2ee5c-a601-412c-8dd9-bf069ecaf1ea id:8129be2d-bd95-412f-a91e-1d94a8c29764 name:ogc-cinjkvvqfsgpnvqf9lr0-3011 type:filebeat version:8.10.0] cloud:map[account:map[id:elastic-platform-ingest] availability_zone:us-central1-a instance:map[id:2851333541445403722 name:ogc-cinjkvvqfsgpnvqf9lr0-3011] machine:map[type:t2a-standard-2] project:map[id:elastic-platform-ingest] provider:gcp region:us-central1 service:map[name:GCE]] component:map[binary:metricbeat dataset:elastic_agent.metricbeat id:system/metrics-default type:system/metrics] data_stream:map[dataset:elastic_agent.metricbeat namespace:q49ily4xofxhfz0rsujmobuotbwk5yer7hfczxcrs type:logs] ecs:map[version:8.0.0] elastic_agent:map[id:8129be2d-bd95-412f-a91e-1d94a8c29764 snapshot:true version:8.10.0] event:map[agent_id_status:verified dataset:elastic_agent.metricbeat ingested:2023-07-12T23:58:15Z] host:map[architecture:aarch64 containerized:false hostname:ogc-cinjkvvqfsgpnvqf9lr0-3011 id:bb1f9231036ddf511f603b0dac92e7bc ip:[10.128.0.51 fe80::4001:aff:fe80:33] mac:[42-01-0A-80-00-33] name:ogc-cinjkvvqfsgpnvqf9lr0-3011 os:map[codename:jammy family:debian kernel:5.19.0-1027-gcp name:Ubuntu platform:ubuntu type:linux version:22.04.2 LTS (Jammy Jellyfish)]] input:map[type:filestream] log:map[file:map[path:/opt/Elastic/Agent/data/elastic-agent-b3f64a/logs/elastic-agent-20230712.ndjson] offset:59544 source:system/metrics-default] log.level:error log.logger:esclientleg log.origin:map[file.line:38 file.name:transport/logging.go] message:Error dialing dial tcp 127.0.0.1:9200: connect: connection refused network:tcp service.name:metricbeat]}]
        	Test:       	TestEnrollAndLog/TestEnroll
    enroll_test.go:84: Cleanup: unenrolling agent
--- FAIL: TestEnrollAndLog/TestEnroll (148.38s)
```

--- a/pkg/testing/tools/check/check.go
+++ b/pkg/testing/tools/check/check.go
@@ -1,0 +1,41 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package check
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/elastic-agent/pkg/control/v2/cproto"
+	integrationtest "github.com/elastic/elastic-agent/pkg/testing"
+)
+
+// ConnectedToFleet checks if the agent defined in the fixture is connected to
+// Fleet Server. It uses assert.Eventually and if it fails the last error will
+// be printed. It returns if the agent is connected to Fleet Server or not.
+func ConnectedToFleet(t *testing.T, fixture *integrationtest.Fixture) bool {
+	t.Helper()
+
+	var err error
+	var agentStatus integrationtest.AgentStatusOutput
+	assertFn := func() bool {
+		agentStatus, err = fixture.ExecStatus(context.Background())
+		return agentStatus.FleetState == int(cproto.State_HEALTHY)
+	}
+
+	connected := assert.Eventually(t, assertFn, 5*time.Minute, 5*time.Second,
+		"want fleet state %s, got %s. agent status: %v",
+		cproto.State_HEALTHY, cproto.State(agentStatus.FleetState), agentStatus)
+
+	if !connected && err != nil {
+		t.Logf("agent isn't connected to fleet-server: last error from agent status command: %v",
+			err)
+	}
+
+	return connected
+}

--- a/testing/integration/monitoring_logs_test.go
+++ b/testing/integration/monitoring_logs_test.go
@@ -2,7 +2,7 @@
 // or more contributor license agreements. Licensed under the Elastic License;
 // you may not use this file except in compliance with the Elastic License.
 
-// //go:build integration
+//go:build integration
 
 package integration
 
@@ -15,49 +15,31 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"github.com/stretchr/testify/suite"
 
 	"github.com/elastic/elastic-agent-libs/kibana"
+	"github.com/elastic/elastic-agent/pkg/control/v2/client"
 	atesting "github.com/elastic/elastic-agent/pkg/testing"
 	"github.com/elastic/elastic-agent/pkg/testing/define"
 	"github.com/elastic/elastic-agent/pkg/testing/tools"
+	"github.com/elastic/elastic-agent/pkg/testing/tools/check"
 )
 
-func TestEnrollAndLog(t *testing.T) {
+func TestMonitoringLogsShipped(t *testing.T) {
 	info := define.Require(t, define.Requirements{
-		OS: []define.OS{
-			{Type: define.Linux},
-		},
+		OS:    []define.OS{{Type: define.Linux}},
 		Stack: &define.Stack{},
 		Local: false,
 		Sudo:  true,
 	})
+	ctx := context.Background()
+
 	t.Logf("got namespace: %s", info.Namespace)
 	t.Skip("Test is flaky; see https://github.com/elastic/elastic-agent/issues/3081")
-	suite.Run(t, &EnrollRunner{requirementsInfo: info})
-}
 
-type EnrollRunner struct {
-	suite.Suite
-	requirementsInfo *define.Info
-	agentFixture     *atesting.Fixture
-}
-
-func (runner *EnrollRunner) SetupSuite() {
-	runner.T().Logf("In SetupSuite")
-	agentFixture, err := define.NewFixture(runner.T(), define.Version())
-	runner.agentFixture = agentFixture
-	require.NoError(runner.T(), err)
-}
-
-func (runner *EnrollRunner) SetupTest() {}
-
-func (runner *EnrollRunner) TestEnroll() {
-	t := runner.T()
-	info := runner.requirementsInfo
-
-	kibClient := info.KibanaClient
+	agentFixture, err := define.NewFixture(t, define.Version())
+	require.NoError(t, err)
 
 	t.Log("Enrolling agent in Fleet with a test policy")
 	createPolicyReq := kibana.AgentPolicy{
@@ -75,32 +57,27 @@ func (runner *EnrollRunner) TestEnroll() {
 			},
 		},
 	}
+
 	// Stage 1: Install
 	// As part of the cleanup process, we'll uninstall the agent
 	installOpts := atesting.InstallOpts{
 		NonInteractive: true,
 		Force:          true,
 	}
-	policy, err := tools.InstallAgentWithPolicy(t, context.Background(), installOpts, runner.agentFixture, kibClient, createPolicyReq)
+	policy, err := tools.InstallAgentWithPolicy(t, ctx,
+		installOpts, agentFixture, info.KibanaClient, createPolicyReq)
 	require.NoError(t, err)
 	t.Logf("created policy: %s", policy.ID)
 
-	t.Cleanup(func() {
-		// After: unenroll
-		t.Logf("Cleanup: unenrolling agent")
-		err = tools.UnEnrollAgent(info.KibanaClient)
-		require.NoError(t, err)
-	})
+	check.ConnectedToFleet(t, agentFixture)
 
-	t.Logf("sleeping for one minute...")
-	time.Sleep(time.Second * 60)
-
-	// Stage 2: check indicies
+	// Stage 2: check indices
 	// This is mostly for debugging
 	resp, err := tools.GetAllindicies(info.ESClient)
 	require.NoError(t, err)
 	for _, run := range resp {
-		t.Logf("%s: %d/%d deleted: %d\n", run.Index, run.DocsCount, run.StoreSizeBytes, run.DocsDeleted)
+		t.Logf("%s: %d/%d deleted: %d\n",
+			run.Index, run.DocsCount, run.StoreSizeBytes, run.DocsDeleted)
 	}
 
 	// Stage 3: Make sure metricbeat logs are populated
@@ -111,21 +88,22 @@ func (runner *EnrollRunner) TestEnroll() {
 	require.NotZero(t, len(docs.Hits.Hits))
 	t.Logf("metricbeat: Got %d documents", len(docs.Hits.Hits))
 
-	// Stage 4: make sure we have no errors
-	t.Log("Making sure there are no error logs")
-	docs = findESDocs(t, func() (tools.Documents, error) {
-		return tools.CheckForErrorsInLogs(info.ESClient, info.Namespace, []string{})
-	})
-	t.Logf("errors: Got %d documents", len(docs.Hits.Hits))
-	for _, doc := range docs.Hits.Hits {
-		t.Logf("%#v", doc.Source)
+	// Stage 4: make sure all components are health
+	t.Log("Making sure all components are healthy")
+	status, err := agentFixture.ExecStatus(ctx)
+	require.NoError(t, err,
+		"could not get agent status to verify all components are healthy")
+	for _, c := range status.Components {
+		assert.Equalf(t, client.Healthy, client.State(c.State),
+			"component %s: want %s, got %s",
+			c.Name, client.Healthy, client.State(c.State))
 	}
-	require.Empty(t, docs.Hits.Hits)
 
 	// Stage 5: Make sure we have message confirming central management is running
 	t.Log("Making sure we have message confirming central management is running")
 	docs = findESDocs(t, func() (tools.Documents, error) {
-		return tools.FindMatchingLogLines(info.ESClient, info.Namespace, "Parsed configuration and determined agent is managed by Fleet")
+		return tools.FindMatchingLogLines(info.ESClient, info.Namespace,
+			"Parsed configuration and determined agent is managed by Fleet")
 	})
 	require.NotZero(t, len(docs.Hits.Hits))
 

--- a/testing/integration/proxy_url_test.go
+++ b/testing/integration/proxy_url_test.go
@@ -17,9 +17,9 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
-	"github.com/elastic/elastic-agent/pkg/control/v2/cproto"
 	integrationtest "github.com/elastic/elastic-agent/pkg/testing"
 	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools/check"
 	"github.com/elastic/elastic-agent/testing/fleetservertest"
 	"github.com/elastic/elastic-agent/testing/proxytest"
 	"github.com/elastic/elastic-agent/version"
@@ -130,7 +130,7 @@ func TestProxyURL_EnrollProxyAndNoProxyInThePolicy(t *testing.T) {
 		require.NoError(t, err, "failed to install agent")
 	}
 
-	p.assertConnectedFleet(t)
+	check.ConnectedToFleet(t, p.fixture)
 }
 
 func TestProxyURL_EnrollProxyAndEmptyProxyInThePolicy(t *testing.T) {
@@ -175,7 +175,7 @@ func TestProxyURL_EnrollProxyAndEmptyProxyInThePolicy(t *testing.T) {
 		require.NoError(t, err, "failed to install agent")
 	}
 
-	p.assertConnectedFleet(t)
+	check.ConnectedToFleet(t, p.fixture)
 }
 
 func TestProxyURL_ProxyInThePolicyTakesPrecedence(t *testing.T) {
@@ -220,7 +220,7 @@ func TestProxyURL_ProxyInThePolicyTakesPrecedence(t *testing.T) {
 		require.NoError(t, err, "failed to install agent")
 	}
 
-	p.assertConnectedFleet(t)
+	check.ConnectedToFleet(t, p.fixture)
 
 	// ensure the agent is communicating through the proxy set in the policy
 	want := fleetservertest.NewPathCheckin(p.policyData.AgentID)
@@ -282,7 +282,7 @@ func TestProxyURL_NoEnrollProxyAndProxyInThePolicy(t *testing.T) {
 		require.NoError(t, err, "failed to install agent")
 	}
 
-	p.assertConnectedFleet(t)
+	check.ConnectedToFleet(t, p.fixture)
 
 	// ensure the agent is communicating through the new proxy
 	if !assert.Eventually(t, func() bool {
@@ -343,7 +343,7 @@ func TestProxyURL_RemoveProxyFromThePolicy(t *testing.T) {
 	}
 
 	// assert the agent is actually connected to fleet.
-	p.assertConnectedFleet(t)
+	check.ConnectedToFleet(t, p.fixture)
 
 	// ensure the agent is communicating through the proxy set in the policy
 	if !assert.Eventually(t, func() bool {
@@ -389,24 +389,7 @@ func TestProxyURL_RemoveProxyFromThePolicy(t *testing.T) {
 	assert.Equal(t, inspect.Fleet.ProxyURL, want)
 
 	// assert, again, the agent is actually connected to fleet.
-	p.assertConnectedFleet(t)
-}
-
-func (p *ProxyURL) assertConnectedFleet(t *testing.T) {
-	t.Helper()
-
-	var err error
-	var agentStatus integrationtest.AgentStatusOutput
-	if !assert.Eventually(t, func() bool {
-		agentStatus, err = p.fixture.ExecStatus(context.Background())
-		return agentStatus.FleetState == int(cproto.State_HEALTHY)
-	}, 5*time.Minute, 5*time.Second,
-		"want fleet state %s, got %s. agent status: %v",
-		cproto.State_HEALTHY, cproto.State(agentStatus.FleetState), agentStatus) {
-		if err != nil {
-			t.Logf("[assertConnectedFleet] last error from agent status command: %v", err)
-		}
-	}
+	check.ConnectedToFleet(t, p.fixture)
 }
 
 func (p *ProxyURL) setupFleet(t *testing.T, fleetHost string) {

